### PR TITLE
Add Strategy form page for account management

### PR DESCRIPTION
**Pull request by CaitBarnard**
## Description of change

Allow users to provide an overview of the strategic direction and long term goals for a company. 

## Test instructions

You should see a form with one input text area field and upon completion you should be redirected to the account management page.
URL is: companies/{companyId}/account-management/strategy/create

### Before

_Add a screenshot_

### After

Screenshot 2023-08-01 at 12 07 42

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [ x ] Has the branch been rebased to main?
- [ x ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ x ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: test/functional/cypress/specs/companies/strategy-spec.js [hunk 1]
```diff
@@ -1,15 +1,55 @@
...
+        assertFieldTextarea({
+          element,
+          label: 'Strategy',
+          hint: "This should outline a plan than provides a concise overview of the business direction and DBT's approach to help them achieve that.",
+        })
+      })
+    })
+
+    it('save strategy button should link to company account management page', () => {
+      cy.get('[data-test="submit-button"]').contains('Save strategy').click()
```
Comment (chopkinsmade) [inline test/functional/cypress/specs/companies/strategy-spec.js:44]: Can you include a test in here for the flash message

**Review by chopkinsmade — APPROVED** [review]

File: package-lock.json
Comment (cgsunkel) [inline package-lock.json:1]: Are these lockfile changes part of the PR?

**Review by cgsunkel — COMMENTED** [review]
I have a few small comments

File: src/client/modules/Companies/AccountManagement/Strategy.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+const Strategy = () => {
+  const { companyId } = useParams()
+  return (
+    <CompanyResource id={companyId}>
+      {(company) => (
+        <DefaultLayout
+          heading={`Add strategy for ${company.name}`}
+          pageTitle={'Strategy'}
+          breadcrumbs={[
```
[the remark this answers is not in the document]
Comment (chopkinsmade) [reply src/client/modules/Companies/AccountManagement/Strategy.jsx:19]: Nice, I didn't know this existed

File: package-lock.json
Comment (backslashbaker) [reply package-lock.json:1]: I've removed the lockfile as it may be a mistake it was added to the PR

**Pull request merged**